### PR TITLE
add cancel button to add-card form

### LINKED ISSUE #677: Add cancel link to the credit-card component
# Problem

We need to add a cancel link centered just below the "Add Card" button on the `credit-card` component.

## Subtasks
- [ ] Ensure link only renders when this is _not_ the first card being created
- [ ] Make the cancel link return you to the card list of existing cards
- [ ] The card form should not have old card information retained if you press cancel
- [ ] Should have a confirmation alert if any of the fields are filled out, so you don't lose all your work
- [ ] Write tests

### PR DESCRIPTION
**Pull request by WenInCode**
# What's in this PR?
Adding cancel button to add-card form.

**This is still in progress**
I still have to add tests, but I am going to sleep right away.
To test up until now, I had added some hardcoded cards to the template and added an add-card button to toggle the `isAddingCard` property. I removed all that for the PR. Will update with tests.

## References
Fixes #677

### CODE REVIEW

File: app/controllers/project/donate.js [hunk 1]
```diff
@@ -3,12 +3,13 @@ import Ember from 'ember';
...
   computed: { alias },
-  inject: { service }
+  inject: { service },
+  set,
 } = Ember;
 
 export default Controller.extend({
   amount: null,
-  isAddingCard: false,
+  isAddingCard: true,
```
Comment (WenInCode) [inline app/controllers/project/donate.js]: I'll need to switch this back, I just switched it for manual testing

File: app/templates/components/donation/credit-card.hbs [hunk 2]
```diff
@@ -40,14 +40,14 @@
...
-  class="button {{unless preventSubmit "default"}} large"
+  class="button {{unless preventSubmit "default"}} large add-card"
   disabled={{preventSubmit}}
   {{action "submit"}}>
   {{#if isSubmitting}}Adding...{{else}}Add Card{{/if}}
 </button>
 
 {{#if canCancel}}
-  <a {{action "cancel"}}>Cancel</a>
+  <button class="button danger large cancel-add-card" {{action "cancel"}}>Cancel</button>
```
Comment (JoshSmith) [inline app/templates/components/donation/credit-card.hbs]: I think this should be a link, not a button, to be less emphasized.

**Comment by WenInCode** [comment]
Updated.
I see there are github has let me know there are some conflicts. Should I just wait to rebase/fix any conflicts?

**Comment by JoshSmith** [comment]
You should rebase now, just don't squash.

**Comment by WenInCode** [comment]
Addressed conflicts

File: app/controllers/project/donate.js [hunk 3]
```diff
@@ -4,7 +4,12 @@ const {
   Controller,
   computed: { alias, bool, not },
   inject: { service },
+<<<<<<< fe46589a16e343be46a8d4732faebe1e61dd2294
```
Comment (WenInCode) [inline app/controllers/project/donate.js]: I do not know how this got in through the rebase... will address right now

**Comment by WenInCode** [comment]
updated.

**Comment by JoshSmith** [comment]
🙌  LGTM.

Squash and rebase and we'll be good.

**Comment by WenInCode** [comment]
Great thanks @JoshSmith

**Comment by JoshSmith** [comment]
❤️ 💜 💚 💛 💙

**Pull request merged**